### DISCUSSION TITLE
Grid Layout with variable row length

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2623] [Q&A]
### Introduction

I'm using GridLayout to automatically layout elements on graph. Currently, I organize the elements in the following manner:
   ```
    const containers = graph.getElements();
    const columns = Math.ceil(Math.sqrt(containers.length));
    layout.GridLayout.layout(containers, {
      columns: columns,
      deep: true,
      columnWidth: 'compact',
      rowHeight: 'compact',
      columnGap: 20,
      rowGap: 20,
    });
```
Now, my requirement is that I need to layout the elements depending on the element type. Let suppose my elements have types 'A', 'B' and 'C'. All the elements of type 'A' must be laid out in first column, all the elements of type 'B' must be laid out in second column and so on.

![image](https://github.com/clientIO/joint/assets/124018900/aadae9f6-78b2-4d15-9c57-41fa2c4a9e08) 

Since the number of rows per each column is variable, is it possible to implement this sort of layout using grid layout?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
No. You need to write a custom function to position the elements like this.
